multimap/src/main/java/org/infinispan/multimap/impl/internal/MultimapObjectWrapper.java

```diff
@@ -77,7 +77,7 @@ public int compareTo(MultimapObjectWrapper other) {
          return Arrays.compare((byte[]) object, (byte[]) other.object);
       }
 
-      return object.toString().compareTo(other.toString());
+      throw new UnsupportedOperationException("can't compare");
```

Comment:
compareTo should return NPE if other == null and ClassCastExc if other cannot be compared